pkg/platform/abstract/project/external/leader/iguazio/synchronizer.go

```diff
@@ -0,0 +1,208 @@
+package iguazio
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/platform/abstract/project"
+	"github.com/nuclio/nuclio/pkg/platform/abstract/project/external/leader"
+	"github.com/nuclio/nuclio/pkg/platformconfig"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+)
+
+type Synchronizer struct {
+	logger                      logger.Logger
+	platformConfiguration       *platformconfig.Config
+	leaderClient                leader.Client
+	internalProjectsClient      project.Client
+	lastSuccessfulSyncTimestamp string
+}
+
+func NewSynchronizer(parentLogger logger.Logger,
+	platformConfiguration *platformconfig.Config,
+	leaderClient leader.Client,
+	internalProjectsClient project.Client) (*Synchronizer, error) {
+
+	parentLogger.DebugWith("Creating project synchronizer")
+	newSynchronizer := Synchronizer{
+		logger:                 parentLogger.GetChild("leader-synchronizer-iguazio"),
+		platformConfiguration:  platformConfiguration,
+		leaderClient:           leaderClient,
+		internalProjectsClient: internalProjectsClient,
+	}
+
+	return &newSynchronizer, nil
+}
+
+func (c *Synchronizer) Start() error {
+	synchronizationInterval, err := time.ParseDuration(c.platformConfiguration.ProjectsLeader.SynchronizationInterval)
+	if err != nil {
+		return errors.Wrap(err, "Failed to parse synchronization interval")
+	}
+
+	// don't synchronize when set to 0
+	if synchronizationInterval == 0 {
+		c.logger.InfoWith("Synchronization interval set to 0. Projects will not synchronize with leader")
+		return nil
+	}
+
+	// start synchronization loop in the background
+	go c.synchronizationLoop(synchronizationInterval)
+
+	return nil
+}
+
+func (c *Synchronizer) synchronizationLoop(interval time.Duration) {
+	c.logger.InfoWith("Starting synchronization loop", "interval", interval)
+
+	ticker := time.NewTicker(interval)
+	for range ticker.C {
+		if err := c.synchronizeProjectsAccordingToLeader(); err != nil {
+			c.logger.WarnWith("Failed to synchronize projects according to leader", "err", err)
+			continue
+		}
+
+		// update last successful sync timestamp
+		c.updateLastSuccessfulSyncTimestamp()
+	}
+}
+
+func (c *Synchronizer) updateLastSuccessfulSyncTimestamp() {
+	loc, err := time.LoadLocation("GMT")
+	if err != nil {
+		c.logger.WarnWith("Failed to load GMT location (Should not happen on unix based systems). "+
+			"Skipping last successful sync timestamp update",
+			"err", err)
+		return
+	}
+
+	t := time.Now().In(loc)
+	c.lastSuccessfulSyncTimestamp = t.Format(time.RFC3339)
```

Comment:
This may create a miss in the synchronization, the timestamp you save and use to query in the next cycle should be the biggest updated_at from the projects returned in the last response

Reply:
Right 👍 I'll change it